File: src/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left =0, right = height.size()-1;
        int area = (right-left)*min(height[left], height[right]);
        while(left<right)
        {
            if(height[left]<height[right])
            {
               left++;
            }
            else
            {
                right--;
            }
            area = max(area, min(height[left], height[right])*(right-left));
        }
        return area;
    }
};

File: python/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height) - 1
        area = (len(height)-1)*min(height[0], height[-1])
        while left<right:
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
            area = max(area, (right-left)*min(height[left], height[right]))
        return area
